.github/workflows/release.yml

```diff
@@ -118,6 +118,65 @@ jobs:
           path: build/release/nvim-macos.tar.gz
           retention-days: 1
 
+  macOS-universal:
+    runs-on: macos-11
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Install brew packages
+        run: |
+          brew update --quiet
+          brew install automake ninja
+      - if: github.event_name == 'push' || (github.event_name == 'workflow_dispatch' && github.event.inputs.tag_name != 'nightly')
+        run: printf 'NVIM_BUILD_TYPE=Release\n' >> $GITHUB_ENV
+      - if: github.event_name == 'schedule' || (github.event_name == 'workflow_dispatch' && github.event.inputs.tag_name == 'nightly')
+        run: printf 'NVIM_BUILD_TYPE=RelWithDebInfo\n' >> $GITHUB_ENV
+      - name: Provision universal `libintl`
+        run: |
+          bottle_tag="arm64_big_sur"
+          brew fetch --bottle-tag="$bottle_tag" gettext
+          cd "$(mktemp -d)"
+          tar xf "$(brew --cache)"/**/*gettext*${bottle_tag}*.tar.gz
+          gettext_prefix="$(brew --prefix gettext)"
+          libintl_install_name="$(otool -D "${gettext_prefix}/lib/libintl.dylib" | sed 1d)"
+          libintl_basename="$(basename "$libintl_install_name")"
+          install_name_tool -id "$libintl_install_name" gettext/*/lib/"$libintl_basename"
+          lipo gettext/*/lib/"$libintl_basename" "$libintl_install_name" -create -output "$libintl_basename"
+          mv -f "$libintl_basename" "$libintl_install_name"
+          codesign --force --sign - "$libintl_install_name"
+      - name: Build release
+        run: |
+          export MACOSX_DEPLOYMENT_TARGET="11"
+          OSX_FLAGS="-DCMAKE_OSX_DEPLOYMENT_TARGET=${MACOSX_DEPLOYMENT_TARGET} -DCMAKE_OSX_ARCHITECTURES=arm64\;x86_64"
+          make CMAKE_BUILD_TYPE=${NVIM_BUILD_TYPE} \
+               CMAKE_EXTRA_FLAGS="-DCMAKE_INSTALL_PREFIX:PATH= $OSX_FLAGS" \
+               DEPS_CMAKE_FLAGS="$OSX_FLAGS"
+          make DESTDIR="$GITHUB_WORKSPACE/build/release/nvim-universal" install
+      - name: Create package
+        run: |
+          cd "$GITHUB_WORKSPACE/build/release"
+          mkdir -p nvim-universal/libs
+          libs=($(otool -L nvim-universal/bin/nvim | sed 1d | sed -E -e 's|^[[:space:]]*||' -e 's| .*||'))
+          echo "libs:"
+          for lib in "${libs[@]}"; do
+            if echo "$lib" | grep -q -E '^(/usr/lib|/System)' 2>/dev/null; then
+              echo "  [skipped] $lib"
+            else
+              echo "  $lib"
+              relname="libs/${lib##*/}"
+              cp -L "$lib" "nvim-universal/$relname"
+              install_name_tool -change "$lib" "@executable_path/../$relname" nvim-universal/bin/nvim
+            fi
+          done
+          codesign --force --sign - nvim-universal/bin/nvim
```

Comment:
Don't we need an Apple developer key or something for this?

Reply:
Nope; that's why we're using ad ad-hoc signature (`--sign -`). That said, this line goes away when I fix the build to link statically to `libintl`, but only because `clang` executes something similar to this under the hood.

Reply:
Another strong argument for static linking; let's just go with that!